certbot/certbot/_internal/renewal.py

```diff
@@ -489,3 +499,13 @@ def handle_renewal_request(config):
     # Windows installer integration tests rely on handle_renewal_request behavior here.
     # If the text below changes, these tests will need to be updated accordingly.
     logger.debug("no renewal failures")
+
+
+def _update_renewal_params_from_key(key_path, config):
+    # type: (str, interfaces.IConfig) -> None
+    with open(key_path, 'rb') as file_h:
+        key = load_pem_private_key(file_h.read(), password=None, backend=default_backend())
+    if isinstance(key, rsa.RSAPrivateKey):
+        config.rsa_key_size = key.key_size
```

Comment:
From https://github.com/certbot/certbot/issues/7711 and the ECDSA design doc, in the future I think we should check if values like `config.rsa_key_size` are set by the user (by checking `certbot.cli.set_by_cli`) and their value differs from the current key and if so raise an error. I think we should do this for ECDSA keys and for `key_type` with RSA keys when support for them is added initially here as I think the only reason we shouldn't do this now with `rsa_key_size` is it is technically a breaking change.

With this in mind, I think it could make sense to check for this case and log a warning for RSA keys now, but I'm also fine if that is done later.